components/style/rule_cache.rs

```diff
@@ -81,12 +82,43 @@ impl RuleCache {
         }
     }
 
+    /// Walk the rule tree and return a rule node for using as the key
+    /// for rule cache.
+    ///
+    /// It currently skips a rule node when it is neither from a style
+    /// rule, nor containing any declaration of reset property. We don't
+    /// skip style rule so that we don't need to walk a long way in the
+    /// worst case. Skipping declarations rule nodes should be enough
+    /// to address common cases that rule cache fails to do reasonable
```

Comment:
You mean `s/rule cache/style sharing cache`? The rule cache should hopefully get a decent sharing now :)

Reply:
I mean, the common case that rule cache failed to do without this...

Reply:
ok, fair, maybe `would fail [...] instead` instead?